actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -90,18 +103,18 @@ class ActionCable.Connection
     open: ->
       ActionCable.log("WebSocket onopen event")
       @disconnected = false
-      @subscriptions.reload()
+      if not @isProtocolSupported()
+        ActionCable.log("WebSocket got '#{@getProtocol()}' as the chosen protocol. Stopping monitor and disconnecting.")
+        @close(allowReconnect: false)
+        return
+      ActionCable.log("WebSocket using '#{@getProtocol()}' subprotocol.")
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
-      @disconnect()
+      return if @disconnected
+      @disconnected = true
```

Comment:
Oh yeah for sure. On every unsuccessful connection this is how the browser says a connection did not happen. I guess it would have made more sense for them to add an event to onError, but what can you do?